packages/config/src/android/Icon.ts

```diff
@@ -1,24 +1,152 @@
+import { resolve } from 'path';
+import Jimp from 'jimp';
+import fs from 'fs-extra';
 import { ExpoConfig } from '../Config.types';
-import { addWarningAndroid } from '../WarningAggregator';
+import * as Colors from './Colors';
+
+type dpiMap = { [dpistring: string]: { folderName: string; scale: number } };
+
+const dpiValues: dpiMap = {
+  mdpi: { folderName: 'mipmap-mdpi', scale: 1 },
+  hdpi: { folderName: 'mipmap-hdpi', scale: 1.5 },
+  xhdpi: { folderName: 'mipmap-xhdpi', scale: 2 },
+  xxhdpi: { folderName: 'mipmap-xxhdpi', scale: 3 },
+  xxxhdpi: { folderName: 'mipmap-xxxhdpi', scale: 4 },
+};
+
+const ANDROID_RES_PATH = 'android/app/src/main/res/';
+const ICON_BACKGROUND = 'iconBackground';
+const IC_LAUNCHER_PNG = 'ic_launcher.png';
+const IC_LAUNCHER_ROUND_PNG = 'ic_launcher_round.png';
+const IC_LAUNCHER_BACKGROUND_PNG = 'ic_launcher_background.png';
+const IC_LAUNCHER_FOREGROUND_PNG = 'ic_launcher_foreground.png';
 
 export function getIcon(config: ExpoConfig) {
-  // Until we add support applying icon config we just test if the user has configured the icon
-  // so we can warn
-  if (config.icon || config.android?.icon) {
-    return true;
-  } else {
-    return false;
-  }
+  return config.icon || config.android?.icon || null;
+}
+
+export function getAdaptiveIcon(config: ExpoConfig) {
+  return {
+    foregroundImage: config.android?.adaptiveIcon?.foregroundImage ?? null,
+    backgroundColor: config.android?.adaptiveIcon?.backgroundColor ?? '#FFFFFF',
+    backgroundImage: config.android?.adaptiveIcon?.backgroundImage ?? null,
+  };
 }
 
 export async function setIconAsync(config: ExpoConfig, projectRoot: string) {
-  let icon = getIcon(config);
+  const { foregroundImage, backgroundColor, backgroundImage } = getAdaptiveIcon(config);
+  let icon = foregroundImage || getIcon(config);
+
   if (!icon) {
-    return;
+    return null;
+  }
+
+  const iconPath = resolve(projectRoot, icon);
+  let length: number;
+
+  // Legacy Icon for Android 7 and earlier
+  for (let dpi in dpiValues) {
```

Comment:
```suggestion
  for (const dpi in dpiValues) {
```
<img width="366" alt="Screenshot 2020-07-02 at 10 12 34" src="https://user-images.githubusercontent.com/5256730/86333746-97e31700-bc4c-11ea-804e-60a7c2476ed4.png">
I know that generally it doesn't matter but I've noticed you're mixing `let` with `const` without explicit consistency.